test/test_OCImain_branch.py

```diff
@@ -0,0 +1,92 @@
+from fixtures import *
+
+"""
+    These tests must only be run when on the main branch
+
+    def get_git_branch(self, file):
+        repo = Repository(file["location"])
+        head = repo.lookup_reference('HEAD').resolve()
+        head = str(head.name).split("/")[-1]
+        return head
+
+        get_git_branch(file) == "main"
+
+"""
+class TestOCIServer:
+    def username_exists(self):
+        return not ((os.environ.get("GHA_USER") is None) or (os.environ.get("GHA_USER") == ""))
+
+    @classmethod
+    def setup_class(cls):
+        pass
+
+    @classmethod
+    def teardown_class(cls):
+        pass
+
+    def test_upload(self, file, powerloader_binary):
+        # Generate a unique file
+        upload_path = generate_unique_file(file)
+
+        # Store the checksum for later
+        hash_before_upload = calculate_sha256(upload_path)
+        print("hash before upload: " + str(hash_before_upload))
+
+        # Upload the file
+        tag, name_on_server = upload_oci(upload_path, powerloader_binary, file["oci_upload_location"])
+
+    def test_download_permanent(self, file, powerloader_binary, checksums):
+        # Delete the file locally
+        # Path(upload_path).unlink()
+
+        # Generate yaml file
+        nos = "artifact"
+        newpath, tmp_yaml = generate_oci_download_yml(file, tag="1.0", name_on_server=nos, username="wolfv")
+        # Download using this YML file
+        download_oci_file(powerloader_binary, tmp_yaml, file)
+
+        # raise Exception("Stop here!")
```

Comment:
can we remove these commetns?

Reply:
Sure :)
the exception was just for debugging...